pkg/apis/pipeline/v1alpha1/pipeline_types.go

```diff
@@ -135,6 +138,15 @@ type PipelineDeclaredResource struct {
 	Type PipelineResourceType `json:"type"`
 }
 
+// PipelineConditionResource allows a Pipeline to declare how its DeclaredPipelineResources
+// should be provided to a Condition as its inputs.
+type PipelineConditionResource struct {
+	// Name is the name of the PipelineResource as declared by the Condition.
+	Name string `json:"name"`
+	// Resource is the name of the DeclaredPipelineResource to use.
```

Comment:
hm this is kinda odd but also really minor - these comments refer to `DeclaredPipelineResource` but I don't think that `DeclaredPipelineResource` is a type that's actually defined anywhere?

Reply:
(Maybe this is `ResourceDeclaration`)?